File: SecureAndroidClient/app/src/main/java/com/security/crypto/IOSocket/IOSynAck.java
package com.security.crypto.IOSocket;

import com.security.crypto.Configuration.CiphersForUse;

import org.json.JSONException;

import java.io.IOException;

public abstract class IOSynAck {

    public abstract void SendPlainMessage() throws IOException, JSONException;

    public abstract void ReceiveServerCertificate() throws Exception;

    public abstract void ResendCookieServer() throws IOException, JSONException;

    public abstract void SendPublicValue() throws IOException;

    public abstract void ReceivePublicValue() throws Exception;

    public abstract void SendCipherSuites() throws Exception;

    public abstract CiphersForUse ReceiveCipherSuites() throws Exception;
}

File: PythonClient/IOSocket/IOSynAck.py
import abc


class SynAck(object):
    __metaclass__ = abc.ABCMeta

    @abc.abstractmethod
    def __SendPlainMessage(self):
        return

    @abc.abstractmethod
    def __ReceiveServerCertificate(self):
        return

    @abc.abstractmethod
    def __ResendCookieServer(self):
        return

    @abc.abstractmethod
    def __SendPublicValue(self):
        return

    @abc.abstractmethod
    def __ReceivePublicValue(self):
        return

    @abc.abstractmethod
    def __SendCipherSuites(self):
        return

    @abc.abstractmethod
    def __ReceiveCipherSuites(self):
        return
